<a href="https://colab.research.google.com/github/KartheekAnusuri/Applied-Tech/blob/main/CapstoneProject4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Instructions

#### Goal of the Project

This project is designed for you to practice and solve the activities that are based on the concepts covered in the following lessons:

1. While Loop, Data-Type Conversion and Conditional Statements.

2. Pandas DataFrame.


---

---

### Activities


  

#### Activity 1: Create Pandas DataFrame

Create a pandas DataFrame by using the below link which has the dataset of NBA players in the CSV format:

**Dataset** : https://media.geeksforgeeks.org/wp-content/uploads/nba.csv

Also, print the first five rows of the dataset.

In [ ]:
# Create a pandas DataFrame
import pandas as pd

# Read the CSV file into a DataFrame
url = "https://media.geeksforgeeks.org/wp-content/uploads/nba.csv"
df = pd.read_csv(url)

# Print the first five rows of the DataFrame
print(df.head())

            Name            Team  Number Position   Age Height  Weight  \
0  Avery Bradley  Boston Celtics     0.0       PG  25.0    6-2   180.0   
1    Jae Crowder  Boston Celtics    99.0       SF  25.0    6-6   235.0   
2   John Holland  Boston Celtics    30.0       SG  27.0    6-5   205.0   
3    R.J. Hunter  Boston Celtics    28.0       SG  22.0    6-5   185.0   
4  Jonas Jerebko  Boston Celtics     8.0       PF  29.0   6-10   231.0   

             College     Salary  
0              Texas  7730337.0  
1          Marquette  6796117.0  
2  Boston University        NaN  
3      Georgia State  1148640.0  
4                NaN  5000000.0  


---

#### Activity 1.2 : Check for Null values

Check for missing values in the DataFrame.

In [ ]:
# Check for missing values in the above Pandas series.
missing_values = df.isnull().sum()
print(missing_values)

Name         1
Team         1
Number       1
Position     1
Age          1
Height       1
Weight       1
College     85
Salary      12
dtype: int64


**Q:** Are there any missing values in the DataFrame ?

**A:** Yes

---

#### Activity 1.3 : Get the Number of rows and columns

Get the number of rows and columns of the DataFrame.

In [ ]:
# Get the Number of rows and columns
df.shape

(458, 9)

**Q:** How many rows are there in the DataFrame?

**A:** 458

---

#### Activity 1.4: Create Pandas series for the player 'Avery Bradley'

Using the `iloc[]` function, create a Pandas series for the player Avery Bradley and store it in a variable called `avery_bradley`.

In [ ]:
# Create pandas series for the player 'Avery Bradley'
avery_bradley = df.iloc[0]
print(avery_bradley)

Name         Avery Bradley
Team        Boston Celtics
Number                 0.0
Position                PG
Age                   25.0
Height                 6-2
Weight               180.0
College              Texas
Salary           7730337.0
Name: 0, dtype: object


**Q**: Which team does Avery Bradely plays for?

**A**: Boston Celtics

---

#### Activity 2: Find the number of solutions for a Quadratic equation

A quadratic equation $ax² + b x + c = 0$ has either 0, 1, or 2 distinct solutions for real values of $x$. Given $a$, $b$ and $c$, you should only return the number of solutions that we can get from the equation.

The above quadratic equation is solved using the formula, \begin{array}{*{20}c} {x = \frac{{ - b \pm \sqrt {b^2 - 4ac} }}{{2a}}} & {{\rm{when}}} & {ax^2 + bx + c = 0} \\ \end{array}

The `±` in the above formula means there are **TWO** answers.

Do you see $b^2 − 4ac$ in the formula above? It is called the **Discriminant**, because it can "discriminate" between the possible types of answer:

  - when $b^2 − 4ac$ is **positive**, we get **two** Real solutions.

  - when $b^2 − 4ac$ is **zero**, we get just **one** real solution.

  - when $b^2 − 4ac$ is **negative**, we get a pair of **complex solution**.


Examples:
```
1. solutions(1, 0, -1) ➞ 2
# x² - 1 = 0 has two solutions (x = 1 and x = -1).

2. solutions(1, 0, 1) ➞ 0
# x² + 1 = 0 has no solutions.
```

**Note**: You do not need to calculate the solutions, just return how many solutions are there by calculating the value of $b^2 − 4ac$.


In [ ]:
# Number of solutions for a quadratic equation
def solutions(a, b, c):
    discriminant = b**2 - 4*a*c

    if discriminant > 0:
        return 2
    elif discriminant == 0:
        return 1
    else:
        return 0

print(solutions(1, 0, -1))
print(solutions(1, 0, 1))

2
0


---